icu4c/source/common/uloc_tag.cpp

```diff
@@ -2124,7 +2124,12 @@ ultag_parse(const char* tag, int32_t tagLen, int32_t* parsedLen, UErrorCode* sta
                 if (*redundantTagEnd  == '\0' || *redundantTagEnd == SEP) {
                     const char* preferredTag = REDUNDANT[i + 1];
                     size_t preferredTagLen = uprv_strlen(preferredTag);
+#if defined(__clang__) || U_GCC_MAJOR_MINOR >= 1100
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstringop-truncation"
                     uprv_strncpy(t->buf, preferredTag, preferredTagLen);
```

Comment:
I think the right fix is
                    uprv_memcpy(t->buf, preferredTag, preferredTagLen);